File: Programmers/위장.py
# 위장

def solution(clothes):
    answer = 1
    clothe = {}
    
    for i in range(len(clothes)):
        if clothes[i][1] not in clothe:
            clothe[clothes[i][1]] = 1
        else:
            tmp = clothe[clothes[i][1]]
            tmp += 1
            clothe[clothes[i][1]] = tmp
            
    for c in clothe:
        answer *= (clothe[c] + 1)
        
    answer -= 1
    
    return answer

File: Programmers/Hash/위장.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, Integer> h = new HashMap<>();
    	
    	for(int i = 0; i < clothes.length; i++) {
    		if(h.get(clothes[i][1]) == null)
    			h.put(clothes[i][1], 1);
    		else
    			h.put(clothes[i][1], h.get(clothes[i][1]) + 1);
    	}
    	
    	for(String keys: h.keySet()) {
    		answer *= (h.get(keys) + 1);
    	}
        
    	answer -= 1;
        return answer;
    }
}